gui/workorder-finetune

```diff
@@ -0,0 +1,544 @@
+--------------------------------------
+local tool_uses = {}
+for i, _ in ipairs(df.tool_uses) do
+    tool_uses[#tool_uses + 1] = df.tool_uses[i]
+end
+local restore_none = false
+if tool_uses[1] == 'NONE' then
+    restore_none = true
+    table.remove(tool_uses, 1)
+end
+table.sort(tool_uses)
+if restore_none then
+    table.insert(tool_uses, 1, 'NONE')
+end
+--------------------------------------
+local set_ore_ix = {}
+for i, raw in ipairs(df.global.world.raws.inorganics) do
+    for _, ix in ipairs(raw.metal_ore.mat_index) do
+        set_ore_ix[ix] = true
+    end
+end
+local ores = {}
+for ix in pairs(set_ore_ix) do
+    local raw = df.global.world.raws.inorganics[ix]
+    ores[#ores+1] = {mat_index = ix, name = raw.material.state_name.Solid}
+end
+table.sort(ores, function(a,b) return a.name < b.name end)
+
+--------------------------------------
+-- CALCIUM_CARBONATE, CAN_GLAZE, FAT, FLUX,
+-- GYPSUM, PAPER_PLANT, PAPER_SLURRY, TALLOW, WAX
+local reaction_classes_set = {}
+for ix,reaction in ipairs(df.global.world.raws.reactions.reactions) do
+    if #reaction.reagents > 0 then
+        for _, r in ipairs(reaction.reagents) do
+            if r.reaction_class and r.reaction_class ~= '' then
+                reaction_classes_set[r.reaction_class] = true
+            end
+        end
+    end --if
+end
+local reaction_classes = {}
+for k in pairs(reaction_classes_set) do
+    reaction_classes[#reaction_classes + 1] = k
+end
+table.sort(reaction_classes)
+--------------------------------------
+-- PRESS_LIQUID_MAT, TAN_MAT, BAG_ITEM etc
+local product_materials_set = {}
+for ix,reaction in ipairs(df.global.world.raws.reactions.reactions) do
+    if #reaction.products > 0 then
+        --for _, p in ipairs(reaction.products) do
+        -- in the list in work order conditions there is no SEED_MAT.
+        -- I think it's because the game doesn't iterate over all products.
+            local p = reaction.products[0]
+            local mat = p.get_material
+            if mat and mat.product_code ~= '' then
+                product_materials_set[mat.product_code] = true
+            end
+        --end
+    end --if
+end
+local product_materials = {}
+for k in pairs(product_materials_set) do
+    product_materials[#product_materials + 1] = k
+end
+table.sort(product_materials)
+--==================================--
+
+local function set_search_keys(choices)
+    for _, choice in ipairs(choices) do
+        if not choice.search_key then
+            if type(choice.text) == 'table' then
+                local search_key = {}
+                for _, token in ipairs(choice.text) do
+                    search_key[#search_key+1] = string.lower(token.text or '')
+                end
+                choice.search_key = table.concat(search_key, ' ')
+            elseif choice.text then
+                choice.search_key = string.lower(choice.text)
+            end
+        end
+    end
+end
+
+guimat.ItemTraitDialog = function(args)
```

Comment:
a script should not be injecting a function into the environment of a library module. You could add it to `gui/materials.lua` in a different PR, though.